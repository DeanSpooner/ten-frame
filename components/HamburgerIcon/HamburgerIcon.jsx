import React from "react";
import { View } from "react-native";
import HamburgerIconStyles from "./HamburgerIcon.styles";

const HamburgerIcon = () => {
  return (
    <View style={HamburgerIconStyles.container}>
      <View style={HamburgerIconStyles.line} />
      <View style={HamburgerIconStyles.line} />
      <View style={HamburgerIconStyles.line} />
    </View>
  );
};

export default HamburgerIcon;
