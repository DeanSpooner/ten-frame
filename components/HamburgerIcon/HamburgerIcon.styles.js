import { StyleSheet } from "react-native";

export default hamburgerIconStyles = StyleSheet.create({
  container: {
    flexDirection: "column",
    margin: 10,
    flex: 1,
    width: 30,
  },
  line: {
    borderColor: "#fff",
    borderRadius: 10,
    borderWidth: 2,
    marginTop: 8,
    width: 35,
  },
  crossLine: {
    borderColor: "#fff",
    borderRadius: 10,
    borderWidth: 2,
    marginTop: 8,
    width: 35,
    transform: [{ rotate: "45deg" }],
  },
});
