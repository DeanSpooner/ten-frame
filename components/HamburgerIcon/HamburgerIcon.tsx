import React from "react";
import { TouchableOpacity, View, Text } from "react-native";
import HamburgerIconStyles from "./HamburgerIcon.styles";

interface HamburgerIconProps {
  setMenuDown: React.Dispatch<React.SetStateAction<boolean>>;
  menuDown: boolean;
}

const HamburgerIcon: React.FC<HamburgerIconProps> = ({
  setMenuDown,
  menuDown,
}) => {
  return (
    <TouchableOpacity
      style={HamburgerIconStyles.container}
      onPress={() => {
        setMenuDown(!menuDown);
      }}
    >
      {!menuDown && (
        <>
          <View style={HamburgerIconStyles.line} />
          <View style={HamburgerIconStyles.line} />
          <View style={HamburgerIconStyles.line} />
        </>
      )}
      {menuDown && (
        <>
          <View style={HamburgerIconStyles.crossLine} />
          <View
            style={[
              HamburgerIconStyles.crossLine,
              { transform: [{ translateY: -12 }, { rotate: "135deg" }] },
            ]}
          />
        </>
      )}
    </TouchableOpacity>
  );
};

export default HamburgerIcon;
