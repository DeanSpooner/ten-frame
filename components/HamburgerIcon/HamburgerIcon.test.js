import React from "react";
import renderer from "react-test-renderer";
import HamburgerIcon from "./HamburgerIcon";

describe("<HamburgerIcon />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<HamburgerIcon />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has 3 children", () => {
    const tree = renderer.create(<HamburgerIcon />).toJSON();
    expect(tree.children.length).toBe(3);
  });
});
