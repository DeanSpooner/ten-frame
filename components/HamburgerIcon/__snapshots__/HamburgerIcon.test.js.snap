// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HamburgerIcon /> renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "column",
      "margin": 10,
      "width": 30,
    }
  }
>
  <View
    style={
      {
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 2,
        "marginTop": 8,
        "width": 35,
      }
    }
  />
  <View
    style={
      {
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 2,
        "marginTop": 8,
        "width": 35,
      }
    }
  />
  <View
    style={
      {
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 2,
        "marginTop": 8,
        "width": 35,
      }
    }
  />
</View>
`;
